operator/controllers/reconcilers/seldon/runtime_reconciler.go

```diff
@@ -40,80 +49,124 @@ func ParseInt32(s string, defaultVal int32) (int32, error) {
 	return int32(i64), err
 }
 
-func ValidateDataflowScaleSpec(
+func getEnvVarValue(podSpec *v1.PodSpec, name string, defaultValue string) string {
+	if podSpec != nil && len(podSpec.Containers) > 0 {
+		for _, env := range podSpec.Containers[0].Env {
+			if env.Name == name {
+				return env.Value
+			}
+		}
+	}
+
+	return defaultValue
+}
+
+func min(a, b int32) int32 {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func replicaCalc(resourceCount, maxConsumers, partitions int32) int32 {
+	if resourceCount == 0 {
+		return 1
+	}
+	return partitions * min(resourceCount, maxConsumers)
+}
+
+func validateScaleSpec(
 	component *mlopsv1alpha1.ComponentDefn,
 	runtime *mlopsv1alpha1.SeldonRuntime,
 	commonConfig common.ReconcilerConfig,
 	namespace *string,
+	envVar string,
```

Comment:
I suggest an argument rename to reflect the fact that `envVar` should contain the max number of consumers.